### PR TITLE
MULE-8221 JmsConnector fails to start when username is configured

### PR DESCRIPTION
**Pull request by dfeist**

### CODE REVIEW

File: transports/jms/src/test/java/org/mule/transport/jms/vendors/ActiveMQJmsConnectorTestCase.java [hunk 1]
```diff
@@ -105,4 +109,23 @@ public void testCustomActiveMqConnectorConfig() throws Exception
...
+
+    /**
+     * See MULE-8221
+     */
+    @Test
+    public void testActiveMqConnectorWithUsernameAndPassword() throws Exception
+    {
+        JmsConnector c = (JmsConnector) muleContext.getRegistry().lookupConnector("activeMqJmsConnectorWithUsernameAndPassword");
+
+        assertTrue(c instanceof ActiveMQJmsConnector);
```
Comment (marianogonzalez) [inline transports/jms/src/test/java/org/mule/transport/jms/vendors/ActiveMQJmsConnectorTestCase.java]: use hamcrest

Comment (marianogonzalez) [inline transports/jms/src/test/java/org/mule/transport/jms/vendors/ActiveMQJmsConnectorTestCase.java]: won't this fail on WMQ? This tests are imported by it and this assertion will fail

File: transports/jms/src/main/java/org/mule/transport/jms/JmsConnector.java [hunk 2]
```diff
@@ -432,7 +432,7 @@ protected Connection createConnection() throws MuleException, JMSException
 
         Connection connection;
 
-        if (username != null)
+        if (!cacheJmsSessions && username != null)
```
Comment (marianogonzalez) [inline transports/jms/src/main/java/org/mule/transport/jms/JmsConnector.java]: what happens if I want to use session caching and still have authentication? Can't I cache the sessions for this user?

File: transports/jms/src/test/java/org/mule/transport/jms/vendors/ActiveMQJmsConnectorTestCase.java (hunk 1 above)
Comment (dfeist) [inline transports/jms/src/test/java/org/mule/transport/jms/vendors/ActiveMQJmsConnectorTestCase.java]: This test is ActiveMQ specific so this isn't an issue.

File: transports/jms/src/main/java/org/mule/transport/jms/JmsConnector.java (hunk 2 above)
Comment (marianogonzalez) [inline transports/jms/src/main/java/org/mule/transport/jms/JmsConnector.java]: how do you prevent wmq from importing this test?

Comment (dfeist) [inline transports/jms/src/main/java/org/mule/transport/jms/JmsConnector.java]: Yes you can, but user/password are set on CachingConnectionFactory and not passed when requesting connection.  This is the cause of this bug.

File: transports/jms/src/test/java/org/mule/transport/jms/vendors/ActiveMQJmsConnectorTestCase.java (hunk 1 above)
Comment (marianogonzalez) [inline transports/jms/src/test/java/org/mule/transport/jms/vendors/ActiveMQJmsConnectorTestCase.java]: how do you prevent wmq from importing this test?

**Pull request merged**